src/kbmod/core/stamp_utils.py

```diff
@@ -331,22 +317,58 @@ def _extract_stamp_stack(imgs, x_vals, y_vals, radius, mask=None):
     Returns
     -------
     stamp_stack : numpy.ndarray
-        A T x S x S sized array where T is the number of times to use
+        A T x (2*R+1) x (2*R+1) sized array where T is the number of times and R is
+        the stamp radius.
+    """
+    num_times = len(imgs)
+    num_stamps = num_times if mask is None else np.count_nonzero(mask)
+    stamp_stack = np.full((num_stamps, 2 * radius + 1, 2 * radius + 1), np.nan)
+
+    # Fill in each unmasked time step.
+    current = 0
+    for idx in range(num_times):
+        if mask is None or mask[idx]:
+            stamp_stack[current] = extract_stamp(imgs[idx], x_vals[idx], y_vals[idx], radius)
+            current += 1
+    return stamp_stack
+
+
+# @jit(nopython=True)
```

Comment:
should this be commented out?

Reply:
No. That was a mistake.  I've removed it.